src/compiler/checker.ts

```diff
@@ -5451,6 +5453,11 @@ namespace ts {
                                 localErrors = false;
                             }
                         }
+                        if (reportErrors) {

```

Comment:
I think this needs to be `if (localErrors) {...}` because you only want to report this error if you haven't reported other errors already. If you look at the changes in all of the tests you'll see what I mean. There are a bunch of "Signature has no corresponding..." errors that are out of place because there actually was a corresponding signature for which you've reported the reason that it wasn't compatible.